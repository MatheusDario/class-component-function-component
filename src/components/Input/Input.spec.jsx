import { fireEvent, screen } from '@testing-library/react';
import { Input } from '.';
import { renderTheme } from '../../styles/render-theme';

const props = {
  value: 'value',
  search: 'value',
};

describe('<Input />', () => {
  it('should have a value of searchValue', () => {
    const mockFuncion = jest.fn();
    renderTheme(
      <Input
        onChange={mockFuncion}
        search={props.search}
        value={props.value}
      />,
    );
    const input = screen.getByPlaceholderText(/Enter a value/i);
    expect(input.value).toBe(props.value);
  });

  it('should call handleChange on each key pressed', () => {
    const mockFuncion = jest.fn();
    renderTheme(
      <Input
        value={props.value}
        search={props.search}
        onChange={mockFuncion}
      />,
    );
    const input = screen.getByPlaceholderText(/Enter a value/i);
    const valueTyping = 'value';

    fireEvent.change(input, { target: { value: valueTyping } });

    expect(input.value).toBe(valueTyping);
  });

  it('should match a snapshot', () => {
    const mockFuncion = jest.fn();
    const { container } = renderTheme(
      <Input
        onChange={mockFuncion}
        search={props.search}
        value={props.value}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
