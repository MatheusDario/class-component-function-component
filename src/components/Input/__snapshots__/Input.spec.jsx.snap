// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match a snapshot 1`] = `
.c1 {
  width: 50%;
  margin: 3rem 0 0 3rem;
  padding: 10px;
}

.c0 {
  color: white;
  margin: 0 0 0 3rem;
}

<div>
  <div
    class=""
  >
    <h1
      class="c0"
    >
      Search Value: 
      value
    </h1>
    <input
      class="c1"
      placeholder="Enter a value"
      type="search"
      value="value"
    />
  </div>
</div>
`;
